contributing/development/compiling/compiling_with_mono.rst

```diff
@@ -8,119 +8,73 @@ Compiling with Mono
-or ``bin/godot.windows.editor.x86_32.mono.exe``. Be especially aware of the **.mono**
-suffix! If you've previously compiled Godot without Mono support, you might have
-similarly named binaries without this suffix. These binaries can't be used to
-generate the Mono glue.
-
-Notes
-^^^^^
-
-- The glue sources must be regenerated every time the ClassDB-registered API
-  changes. That is, for example, when a new method is registered to the
-  scripting API or one of the parameters of such a method changes.
-  Godot will print an error at startup if there is an API mismatch
-  between ClassDB and the glue sources.
-
-
-Rebuild with Mono glue
-----------------------
-
-Once you have generated the Mono glue, you can generate the final binary with
+This command will tell Godot to generate the C# bindings for the Godot API at
+``modules/mono/glue/GodotSharp/GodotSharp/Generated``, and the C# bindings for
+the editor tools at ``modules/mono/glue/GodotSharp/GodotSharpEditor/Generated``.
+Once these files are generated, you can build Godot's managed libraries for all
+the desired targets without having to repeat this process.
+
+``<godot_binary>`` refers to the editor binary you compiled with the Mono module
+enabled. Its exact name will differ based on your system and configuration, but
+should be of the form ``bin/godot.<platform>.editor.<arch>.mono``, e.g.
+``bin/godot.linuxbsd.editor.x86_64.mono`` or 
+``bin/godot.windows.editor.x86_32.mono.exe``. Be especially aware of the
+**.mono** suffix! If you've previously compiled Godot without Mono support, you
+might have similarly named binaries without this suffix. These binaries can't be
+used to generate the Mono glue.
+
+.. note:: The glue sources must be regenerated every time the ClassDB-registered
+          API changes. That is, for example, when a new method is registered to
+          the scripting API or one of the parameters of such a method changes.
+          Godot will print an error at startup if there is an API mismatch
+          between ClassDB and the glue sources.
+
+Building the managed libraries
+------------------------------
+
+Once you have generated the Mono glue, you can builds the managed libraries with
 the ``build_assemblies.py`` script.::
 
-    ./modules/mono/build_scripts/build_assemblies.py --godot-output-dir=./bin --godot-platform={PLATFORM_NAME}
-
-And Mono-enabled export templates::
+    ./modules/mono/build_scripts/build_assemblies.py --godot-output-dir=./bin
 
-    scons p=<platform> target=template_release module_mono_enabled=yes
+If everything went well, the ``GodotSharp`` directory, containing the managed
+libraries, should have been created in the in the ``bin`` directory.
 
-If everything went well, apart from the normal output, SCons should have created
-the following files in the ``bin`` directory:
+.. note:: Due to NuGet package caching, rebuilding the managed libraries does
+          not reliably update all projects using them. To solve this issue
```

Comment:
I don't like the phrasing in here, it makes it sound as if it's a NuGet problem when in reality this behavior is entirely our doing in Godot since we copy the NuGet packages when building (but only if packages with the same version don't already exist in the directory).

We plan to get rid of the `GodotNuGetFallbackFolder` anyway, I think we were going to do that with _editor unification_ but in the end that didn't make it for 4.0.

Reply:
This is not entirely a godot specific issue, I had issues with other packages as well. TBH I am actually pretty surprised that deleting `GodotNuGetFallbackFolder` even works and that the copy in `~/.nuget/packages` that nuget makes doesn't cause any issues.

Reply:
I haven't had issues with other packages, but if it's resolved by deleting `GodotNuGetFallbackFolder` then it seems specific to Godot, since that folder should only contain Godot packages.

Keep in mind that the configuration we add to `~/.config/NuGet/config/Godot.Offline.Config` affects non-Godot projects as well, when restoring any NuGet package (including non-Godot packages).

Reply:
I left this as is for now, I'll have to test some things again and come back to this.

Reply:
[nuget_local_package_cache.zip](https://github.com/godotengine/godot-docs/files/10785701/nuget_local_package_cache.zip)
This demonstrates the issue I was talking about in a fresh docker container that has never seen anything godot:
1. build the docker container
2. start it and run `/test/run.sh`
   - it builds a package and publishes it to a local nuget source
   - runs a project showing the old message
   - updates the package source to a new message
   - rebuilds the package and publishes it again
   - re-runs the project still showing the old message
3. the new message will be shown on re-run if the package is linked as a project reference instead or if `~/ .nuget/packages` is cleared between the runs.

So while the exact parameters of this issue are godot specific, the same symptoms can also happen when no using godot.

Also I found why the nugets cache folder doesn't cause problems: The build is actually setup to delete the package being built from the nuget cache to avoid this issue, but that bit just doesn't do the same for godots cache, and thus not actually solving anything.